Source/ARTRealtime.m

```diff
@@ -712,13 +730,8 @@ - (ARTEventListener *)performTransitionWithStateChange:(ARTConnectionStateChange
         case ARTRealtimeConnected: {
             _fallbacks = nil;
             _connectionLostAt = nil;
-            if (stateChange.reason) {
-                ARTStatus *status = [ARTStatus state:ARTStateError info:[stateChange.reason copy]];
-                [self failPendingMessages:status];
-            }
-            else {
-                [self resendPendingMessages];
-            }
+            self.options.recover = nil; // RTN16k
+            [self resendPendingMessagesWithResumed:stateChange.resumed];
```

Comment:
What I meant by this is, you should move 
```
            self.options.recover = nil; // RTN16k
            [self resendPendingMessagesWithResumed:stateChange.resumed];
```
right below this line

https://github.com/ably/ably-cocoa/blob/415810bf9347dc876d19cc38c12efccdd2bd955a/Source/ARTRealtime.m#L886-L897

This way you can directly access `resumed` variable from https://github.com/ably/ably-cocoa/blob/415810bf9347dc876d19cc38c12efccdd2bd955a/Source/ARTRealtime.m#L886

Reply:
Then you don't really have to set resumed flag for connStateChange. So, you can remove metadata varibale from https://github.com/ably/ably-cocoa/blob/415810bf9347dc876d19cc38c12efccdd2bd955a/Source/ARTRealtime.m#L906C14-L906C14

Reply:
Yes, sure I can move this call to use resumed var directly. But since metadata is a helper object to pass additional params, why not to use it here? WDYT @lawrence-forooghian

Reply:
I feel right place to use 
```
            self.options.recover = nil; // RTN16k
            [self resendPendingMessagesWithResumed:stateChange.resumed];
```
is where resume is detected. Though I agree metadata is available, we don't really need to force use it.
Code simplicity in terms of readability is important. We are doing the same in other SDK's.

https://github.com/ably/ably-go/blob/aea1a51ea5e67653453304778ff3f26f93cfff90/ably/realtime_conn.go#L851

https://github.com/ably/ably-dotnet/blob/c2a514ad2cc9401547c8c6ccb7c23b87101131a5/src/IO.Ably.Shared/Realtime/Workflows/RealtimeWorkflow.cs#L625-L643

Note - If metadata is only being used to set `resume` flag, then we can remove `metadata` itself altogether. Also, it will be better if we include those changes as a part of this PR itself.

Reply:
I don't feel like this change is necessary, lets wait for @lawrence-forooghian opinion on this. The idea was to keep connection setup in `onConnected:` and all additional code in `[self transition:ARTRealtimeConnected withMetadata:metadata];`. Moving that line breaks it for no substantial reason.

Reply:
Yeah, personally I'd prefer to avoid having to restructure the code more than is necessary, so I'd agree with @maratal. But I'll leave it to you two to decide. It's something we could change later on, right?

Reply:
I've created an issue for this  https://github.com/ably/ably-cocoa/issues/1856

Reply:
Thanks, I was expecting this to be addressed as a part of this PR itself though : (

Reply:
It's not a necessary change for this particular PR. Also I would argue if it's relevant without refactoring the whole code in `onConnected:`. If you move only this line, why not others then? Why not to just get rid of `transition:` completely?

Reply:
Yeah, I would agree to get rid of `transition:` itself 👍 . I think it's always best to refactor the code when you have understanding of the code context. 
I would let you decide on that part, provided this will be taken care of in the future : )

Reply:
I agree that the code might be confusing for the new developer, but once you get used to it, it's actually not that bad for objective-c codebase (I've seen way worse than that). In any case it's a very deep refactoring since `transition:` is used heavily here and for sure shouldn't be a part of this PR - https://github.com/ably/ably-cocoa/issues/1858